From 30ea3122ce187039bf516825c0970e314fdf90d6 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Fri, 25 Mar 2022 11:23:35 -0300
Subject: [PATCH 135/163] posix: Suppress clang warning on bug-regex24

clang complains that adding a 'regoff_t' to a string doe snot
append to it, and the printf idea below is to make rm_so an
offset.
---
 posix/bug-regex24.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/posix/bug-regex24.c b/posix/bug-regex24.c
index 97c5c3508a..7a655cbc4d 100644
--- a/posix/bug-regex24.c
+++ b/posix/bug-regex24.c
@@ -1,6 +1,7 @@
 #include <regex.h>
 #include <stdio.h>
 #include <string.h>
+#include <libc-diag.h>
 
 #define str "civic"
 
@@ -45,11 +46,17 @@ do_test (void)
       {
 	int len = m[i].rm_eo - m[i].rm_so;
 
+	/* clang complains that adding a 'regoff_t' to a string does not
+	   append to it, and the printf idea below is to make rm_so as
+	   an offset to str.  */
+	DIAG_PUSH_NEEDS_COMMENT_CLANG;
+	DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wstring-plus-int");
 	printf ("m[%d] = \"%.*s\"\n", i, len, str + m[i].rm_so);
 
 	if (strlen (expected[i]) != len
 	    || memcmp (expected[i], str + m[i].rm_so, len) != 0)
 	  result = 1;
+	DIAG_POP_NEEDS_COMMENT_CLANG;
       }
 
   return result;
-- 
2.37.2

